shotover-proxy/src/codec/cassandra.rs

```diff
@@ -1,23 +1,25 @@
+use crate::codec::{Codec, CodecReadError};
 use crate::frame::cassandra::{CassandraMetadata, CassandraOperation, Tracing};
 use crate::frame::{CassandraFrame, Frame, MessageType};
 use crate::message::{Encodable, Message, Messages, Metadata};
-use crate::server::CodecReadError;
 use anyhow::{anyhow, Result};
 use bytes::{Buf, BufMut, BytesMut};
 use cassandra_protocol::compression::Compression;
 use cassandra_protocol::frame::message_error::{ErrorBody, ErrorType};
-use cassandra_protocol::frame::message_supported::BodyResSupported;
+use cassandra_protocol::frame::message_startup::BodyReqStartup;
 use cassandra_protocol::frame::{
-    CheckEnvelopeSizeError, Envelope as RawCassandraFrame, Opcode, Version,
+    CheckEnvelopeSizeError, Envelope as RawCassandraFrame, Flags, Opcode, Version,
 };
 use cql3_parser::cassandra_statement::CassandraStatement;
 use cql3_parser::common::Identifier;
+use std::sync::Arc;
+use std::sync::RwLock;
 use tokio_util::codec::{Decoder, Encoder};
 use tracing::info;
 
 #[derive(Debug, Clone)]
 pub struct CassandraCodec {
-    compression: Compression,
+    compression: Arc<RwLock<Compression>>,
```

Comment:
Given we would likely just set this once in a session (unless the spec says otherwise), do we need the Arc<RwLock<>>? This is just my gut feel from an eyeball read, the compiler sounds like it says something different otherwise?

Reply:
The compression won't be changed but this is needed because the agreed compression needs to be shared to both the read and write codec.